openedx/license-manager
# ENT-3458: Implement License revocation cap for SubscriptionPlan

[ENT-3458](https://openedx.atlassian.net/browse/ENT-3458)

## Description

* added fields to SubscriptionPlan model for tracking revoke usage
* added revocation data to SubscriptionPlanSerializer
* only revoke if revocations remain on the SubscriptionPlan, return 400 otherwise

## Testing Instructions

- Checkout branch, run migrations, verify that activated licenses exist for a given customer. 
- Login to Admin Portal and revoke a license
- Observe the number of revocations remaining decrease on the SubscriptionPlan in Django Admin
- Set 'Revoke max percentage' to 0 on your SubscriptionPlan
- Attempt revoking a license again from the Admin Portal. Observe a request failure with status code 400.

## Post-review

Squash commits into discrete sets of changes
